vending-machine/readme.md

```diff
@@ -0,0 +1,114 @@
+## Introduction
+This guide will walk you through the process of building contracts for the vending machine, deploying them into an existing world running in Docker, and testing their functionality by executing scripts.
+
+The vending machine allows you to automatically trade an item for another item. For example, if a player gives you one set item depending on the ratio you set it could give the player two of another item.
+
+## Deployment and Testing
+### Step 0: Deploy the vending machine contracts to the existing world 
+First, copy the World Contract Address from the Docker logs obtained in the previous step, then run the following command:
+
+![alt text](../docker_deployment.png)
+
+```bash
+cd vending-machine/packages/contracts
+```
+
+Install the dependecies for the contracts:
+```bash
+pnpm install
+```
+
+**Local Deployment**
+This will deploy the contracts to your local world.
+```bash
+pnpm run deploy:local --worldAddress <worldAddress> 
+```
+
+**Devnet Deployment**
+This will deploy the contracts to the Devnet world.
+```bash
+cd packages/contracts
+pnpm run deploy:devnet --worldAddress <worldAddress> 
+```
+
+
+eg: `pnpm run deploy:local --worldAddress 0xafc8e4fd5eee66590c93feebf526e1aa2e93c6c3`
+
+Once the deployment is successful, you'll see a screen similar to the one below. This process deploys the Vending Machine contract.
+![alt text](./readme-imgs/deployment.png)
+
+
+### Step 1: Setup the environment variables 
+Next, replace the following values in the [.env](./packages/contracts/.env) file with the values you copied earlier:
+
+```bash
+#WORLD ADDRESS COPIED FROM DOCKER LOGS
+WORLD_ADDRESS=
+
+#DONT NEED TO CHANGE IF YOUR RUNNING LOCALLY
+SSU_ID=
+```
+
+You can adjust the remaining values in the .env file as needed, though they are optional.
+
+<details markdown="block">
+<summary>Customizing optional environment values</summary>
+
+#### Items
+You can set the items in and out to change which items you trade for. 
+
+```bash
+#ITEM ID 77800 - Common Ore
+ITEM_IN_ID=888
```

Comment:
we should add details where they could fetch these values in game 
https://github.com/projectawakening/builder-examples/blob/stable-release/README.md#step-2-env-changes

We should probably do this for all .env file. To say where and how they can these values

Reply:
Have added to all of the environment files.